org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ProjectEncodingMode.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+/**
+ *
+ * @author snjeza
+ *
+ */
+public enum ProjectEncodingMode {
+	IGNORE, /* Ignore project encoding settings */
```

Comment:
This is fine, so we can keep it.

Just an observation. I noticed is that it is possible to define something like `IGNORE("ignore")` , and combined with a constructor definition, it eliminates the need for the conversion logic. This is fine though also. Arguably having to re-define the constants as lower case here is not as nice as the toUpperCase approach.

Reply:
See also https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InlayHintsParameterMode.java#L17

Reply:
Yup, I saw a few of these which is what made me decide it's fine.